test_unstructured_ingest/dest/pinecone.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
+set -e
+
+DEST_PATH=$(dirname "$(realpath "$0")")
+SCRIPT_DIR=$(dirname "$DEST_PATH")
+cd "$SCRIPT_DIR"/.. || exit 1
+OUTPUT_FOLDER_NAME=s3-pinecone-dest
+OUTPUT_DIR=$SCRIPT_DIR/structured-output/$OUTPUT_FOLDER_NAME
+WORK_DIR=$SCRIPT_DIR/workdir/$OUTPUT_FOLDER_NAME
+DOWNLOAD_DIR=$SCRIPT_DIR/download/$OUTPUT_FOLDER_NAME
+max_processes=${MAX_PROCESSES:=$(python3 -c "import os; print(os.cpu_count())")}
+
+if [ -z "$PINECONE_API_KEY" ]; then
+   echo "Skipping Pinecone ingest test because PINECONE_API_KEY env var is not set."
+   exit 0
+fi
+
+RANDOM_SUFFIX=$((RANDOM % 100000 + 1))
+
+PINECONE_ENVIRONMENT="us-east1-gcp"
+PINECONE_INDEX="ingest-test-$RANDOM_SUFFIX"
+PINECONE_PROJECT_ID="art8iaj"
+
+# shellcheck disable=SC1091
+source "$SCRIPT_DIR"/cleanup.sh
+function cleanup {
+
+  # Get response code to check if index exists
+  response_code=$(curl \
+    -s -o /dev/null \
+    -w "%{http_code}" \
+    --request GET \
+    --url https://controller.$PINECONE_ENVIRONMENT.pinecone.io/databases/$PINECONE_INDEX \
+    --header 'accept: application/json' \
+    --header "Api-Key: $PINECONE_API_KEY")
+
+  # Cleanup (delete) index if it exists
+  if [ "$response_code" == "200" ]; then
+    echo ""
+    echo "deleting index $PINECONE_INDEX"
+    curl --request DELETE \
+    "https://controller.$PINECONE_ENVIRONMENT.pinecone.io/databases/$PINECONE_INDEX" \
+    --header "Api-Key: $PINECONE_API_KEY" \
+    --header 'content-type: application/json'
+
+  else
+    echo "Index $PINECONE_INDEX does not exist, nothing to delete"
+  fi
+
+  # Local file cleanup
+  cleanup_dir "$WORK_DIR"
+  cleanup_dir "$OUTPUT_DIR"
+  if [ "$CI" == "true" ]; then
+    cleanup_dir "$DOWNLOAD_DIR"
+  fi
+}
+
+trap cleanup EXIT
+
+echo "Creating index $PINECONE_INDEX"
+create_attempt=0
+create_sleep_amount=10
+response_code=999
+
+while [ "$response_code" -ge 400 ] && [ "$create_attempt" -lt 3 ]; do
+  create_attempt=$((create_attempt+1))
+  echo "attempt $create_attempt for index creation"
+  response_code=$(curl \
+     -s -o /dev/null \
+     -w "%{http_code}" \
+     --request POST \
+     --url "https://controller.$PINECONE_ENVIRONMENT.pinecone.io/databases" \
+     --header "accept: text/plain" \
+     --header "content-type: application/json" \
+     --header "Api-Key: $PINECONE_API_KEY" \
+     --data '
+{
+  "name": "'"$PINECONE_INDEX"'",
+  "dimension": 384,
+  "metric": "cosine",
+  "pods": 1,
+  "replicas": 1,
+  "pod_type": "p1.x1"
+}
+')
+
+  if [ "$response_code" -ge 400 ]; then
+  echo "Response code: $response_code . Sleeping $create_sleep_amount seconds to retry index creation."
```

Comment:
Just curious, why would the index creation itself cause issue and require a wait? I could see us needing to wait for the index to be up and ready after the initial create request.

Reply:
Left from when we had one index limit - just tested this and it does work with multiple tests running. Removed with: [remove download_dir, remove index creation loop](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/387a7ad9c88f171ac3b636fc40a75aea7a6f08f3)